velox/connectors/hive/iceberg/tests/PartitionSpecTest.cpp

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/connectors/hive/iceberg/PartitionSpec.h"
+
+#include <gtest/gtest.h>
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/prestosql/types/TimestampWithTimeZoneType.h"
+#include "velox/type/Type.h"
+
+namespace facebook::velox::connector::hive::iceberg {
+
+namespace {
+
+TEST(PartitionSpecTest, invalidColumnType) {
+  std::vector<IcebergPartitionSpec::Field> fields1 = {
+      {"col_row",
+       ROW({{"a", INTEGER()}}),
+       TransformType::kIdentity,
+       std::nullopt},
+  };
+  VELOX_ASSERT_USER_THROW(
+      std::make_shared<const IcebergPartitionSpec>(1, fields1),
+      "Type is not supported as a partition column: ROW");
+
+  std::vector<IcebergPartitionSpec::Field> fields2 = {
+      {"col_array", ARRAY(INTEGER()), TransformType::kIdentity, std::nullopt},
+  };
+  VELOX_ASSERT_USER_THROW(
+      std::make_shared<const IcebergPartitionSpec>(1, fields2),
+      "Type is not supported as a partition column: ARRAY");
+
+  std::vector<IcebergPartitionSpec::Field> fields3 = {
```

Comment:
add a helper lambda to avoid copy-paste; column name and transform type can be hard-coded since these are all the same

```
auto makeSpec = [] (type) {
   return ...
};

VELOX_ASSERT_USER_THROW(
      makeSpec(MAP(...)),
      "Type is not supported as a partition column: MAP");
```